android/src/main/kotlin/com/signify/hue/flutterreactiveble/ble/BleWrapper.kt

```diff
@@ -3,7 +3,31 @@ package com.signify.hue.flutterreactiveble.ble
 import com.polidea.rxandroidble2.RxBleConnection
 import java.util.UUID
 
-data class ScanInfo(val deviceId: String, val name: String, val rssi: Int, val serviceData: Map<UUID, ByteArray>)
+data class ScanInfo(val deviceId: String, val name: String, val rssi: Int, val serviceData: Map<UUID, ByteArray>, val manufacturerData: ByteArray) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as ScanInfo
```

Comment:
Won't this blow, if `other` is not `ScanInfo`? Is that a convention?

I'd expect `1 == "text"` to be false or rejected at compile time.

Reply:
This code was generated by IntelliJ, I will have a look.

Reply:
Line 8 checks that other is the same class.
The Kotlin linter warns about a "Array property in data class: it's recommended to override equals() / hashCode()" and provide this recommended implementation.

Reply:
Ah, it's generated code. I guess, we can trust IDEA 🙂 Thanks for checking.